js/utils.js

```diff
@@ -1300,12 +1301,20 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
           field: field,
           useData: true
         }), function(item) {
+          var dates = records.map(function(obj) {
+            return moment(obj.data.Date);
+          });
+          var minDate = moment.min(dates).format(LOCAL_FORMAT);
+          var maxDate = moment.max(dates).format(LOCAL_FORMAT);
           var classConverted = _.kebabCase(item);
           var newObj = {
             type: field,
             data: {
               name: item,
-              class: classConverted
+              class: classConverted,
+              type: field,
+              minDate: minDate,
```

Comment:
I don't understand why every single record in the data set needs a `minDate` and `maxDate` property. Each date filter probably needs these properties, but not each record.